workers/data_refinery_workers/processors/smashing_utils.py

```diff
@@ -369,8 +373,16 @@ def process_frames_for_key(key: str,
                 elif frame['technology'] == 'rnaseq':
                     rnaseq_columns.extend(frame['dataframe'].columns)
 
-        all_gene_identifiers = list(all_gene_identifiers)
+        total_samples = len(microarray_columns) + len(rnaseq_columns)
+        all_gene_identifiers = []
+        for gene_id, sample_count in gene_identifier_counts.items():
+            # We only want to use gene identifiers which are present
+            # in >50% of the samples.
```

Comment:
It would probably be good to note that we're doing this because a large number of gene identifiers present in only a modest number of experiments have leaked through. We wouldn't necessarily _want_ to do this if we'd mapped all the data to ENSEMBL identifiers successfully.